code/modules/mob/living/simple_animal/abnormality/he/golden_false_apple.dm

```diff
@@ -0,0 +1,420 @@
+	QDEL_IN(src, 10 SECONDS)
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/golden_apple/proc/BecomeRotten()//phase 2
+	if(QDELETED(src))
+		return
+	if(is_maggot)//prevents the proc from being spammed
+		return
+	playsound(src, 'sound/abnormalities/goldenapple/Gold_Falsify.ogg', 50, 1)//it's very loud
+	icon = 'ModularTegustation/Teguicons/96x48.dmi'
+	icon_state = "false_apple"
+	icon_living = "false_apple"
+	icon_dead = "false_egg"
+	deathmessage = "is reduced to a primoridal egg."
+	name = "False Apple"
+	desc = "The apple ruptured and a swarm of maggots crawled inside, metamorphosing into a hideous face."
+	pixel_x = -48
+	pixel_y = 0
+	light_range = 0
+	light_power = 0
+	attack_sound = list('sound/abnormalities/goldenapple/False_Attack.ogg', 'sound/abnormalities/goldenapple/False_Attack2.ogg')
+	melee_damage_lower = 30
+	melee_damage_upper = 45
+	attack_verb_continuous = "pummels"
+	attack_verb_simple = "pummel"
+	move_to_delay = 3
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.5, WHITE_DAMAGE = 1.5, BLACK_DAMAGE = 1, PALE_DAMAGE = 0.5)
+	melee_damage_type = BLACK_DAMAGE
+	fear_level = WAW_LEVEL
+	revive(full_heal = TRUE, admin_revive = FALSE)
+	is_maggot = TRUE
+
+//2-tile attack range to take advantage of its AOE
+/mob/living/simple_animal/hostile/abnormality/golden_apple/CheckAndAttack()//taken from porccubus
+	if(!target)
+		return
+	if(targets_from && isturf(targets_from.loc) && get_dist(target, src) <= 2 && !incapacitated())//Only extends the range for the Smash() proc
+		AttackingTarget()
+		return
+
+/mob/living/simple_animal/hostile/abnormality/golden_apple/AttackingTarget()
+	if(!can_act)
+		return FALSE
+	if(!is_maggot)
+		return ..()
+	if(prob(50))
+		return ..()
+	Smash(target)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/golden_apple/proc/EatEmployees()
+	var/last_target
+	var/target_hit
+	can_act = FALSE
+	playsound(get_turf(src), 'sound/abnormalities/goldenapple/False_Attack2.ogg', 100, 0, 5)
+	for(var/turf/T in view(1, src))
+		new /obj/effect/temp_visual/smash_effect(T)
+		for(var/mob/living/L in T)
+			if(faction_check_mob(L))
+				continue
+			L.apply_damage(200, RED_DAMAGE, null, L.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)//as powerful as goodbye, but only a 1-tile radius
+			if(L.health < 0)
+				L.gib()
+				if(!last_target)//only the last person killed counts
+					addtimer(CALLBACK(src, .proc/DigestPerson, L), 5 SECONDS)
+					last_target = TRUE
+					target_hit = TRUE
+		if (!target_hit)
+			addtimer(CALLBACK(src, .proc/BecomeRotten), 5 SECONDS)//if nobody got killed
+	can_act = TRUE
+
+/mob/living/simple_animal/hostile/abnormality/golden_apple/proc/DigestPerson(mob/living/carbon/human/H)//berserk mode
+	BecomeRotten()
+	maxHealth = 1500
+	src.adjustBruteLoss(-300)
+	med_hud_set_health()//took a page from smock to update medhuds
+	med_hud_set_status()
+	update_health_hud()
+	move_to_delay = 2.5//same as awakened blue smock
+	revive(full_heal = TRUE, admin_revive = FALSE)
```

Comment:
I've come to the realization you call revive here, with the full heal arg set to `TRUE`, meaning line 300 is pointless here